src/main/kotlin/org/rust/ide/intentions/GenerateDocstringStubIntention.kt

```diff
@@ -0,0 +1,91 @@
+        val targetFunc = element.ancestorOrSelf<RsGenericDeclaration>() ?: return null
+        if (targetFunc !is RsFunctionImpl) return null
+        if (targetFunc.name != element.text) return null
+        if (targetFunc.text.startsWith("///")) return null
+        val params = targetFunc.valueParameters
+        if (params.isEmpty()) {
+            return null
+        }
+        val returnType = targetFunc.returnType
+        return Context(targetFunc, params, returnType)
+    }
+
+    override fun invoke(project: Project, editor: Editor, ctx: Context) {
+        val (targetFunc, params, returnType) = ctx
+        val document = editor.document
+        var commentStartOffset: Int = targetFunc.textRange.startOffset
+        val lineStartOffset = document.getLineStartOffset(document.getLineNumber(commentStartOffset))
+        if (lineStartOffset > 0 && lineStartOffset < commentStartOffset) {
+            val nonWhiteSpaceOffset = CharArrayUtil.shiftBackward(document.charsSequence, commentStartOffset - 1, " \t")
+            commentStartOffset = Math.max(nonWhiteSpaceOffset, lineStartOffset)
+        }
+        val buffer = StringBuilder()
+        var commentBodyRelativeOffset = 0
+        val indentOffset = targetFunc.startOffset - lineStartOffset
+        val indentation = " ".repeat(indentOffset)
+        buffer.append("$indentation/// \n")
+        commentBodyRelativeOffset += buffer.length
+
+        document.insertString(commentStartOffset, buffer)
+        val docManager = PsiDocumentManager.getInstance(project)
+        docManager.commitDocument(document)
+        val stub = generateDocumentStub(indentation, params, returnType)
+        if (stub.isNotEmpty()) {
+            val insertionOffset = commentStartOffset + commentBodyRelativeOffset
+            document.insertString(insertionOffset, stub)
+            docManager.commitDocument(document)
+        }
+        editor.caretModel.moveToOffset(targetFunc.startOffset + buffer.length - indentOffset - 1)
+    }
+}
+
+
+private fun generateDocumentStub(indentation: String, params: List<RsValueParameter>, returnType: Ty): String {
+    val builder = StringBuilder()
+    builder.append("$indentation/// \n")
+    builder.append("$indentation/// # Arguments \n")
+    builder.append("$indentation/// \n")
+    var paramName: String?
+    for (param in params) {
+        paramName = param.patText
+        builder.append("$indentation/// * `$paramName`: \n")
+    }
+    builder.append("$indentation/// \n")
+    if (returnType != null) {
+        builder.append("$indentation/// returns: $returnType \n")
+    }
+    builder.append("$indentation/// \n")
+    builder.append("$indentation/// # Examples \n")
+    builder.append("$indentation/// \n")
+    builder.append("$indentation/// ```\n")
+    builder.append("$indentation/// \n")
+    builder.append("$indentation/// ```\n")
+    return builder.toString()
+}
```

Comment:
```suggestion
private fun generateDocumentStub(
    indentation: String,
    params: List<RsValueParameter>,
    returnType: Ty
): String = buildString {
    append("$indentation/// \n")
    append("$indentation/// # Arguments \n")
    append("$indentation/// \n")
    for (param in params) {
        append("$indentation/// * `${param.patText}`: \n")
    }
    append("$indentation/// \n")
    if (returnType != null) {
        append("$indentation/// returns: $returnType \n")
    }
    append("$indentation/// \n")
    append("$indentation/// # Examples \n")
    append("$indentation/// \n")
    append("$indentation/// ```\n")
    append("$indentation/// \n")
    append("$indentation/// ```\n")
}
```